openedx/core/djangoapps/certificates/api.py

```diff
@@ -0,0 +1,43 @@
+"""
+The public API for certificates.
+"""
+
+from openedx.core.djangoapps.certificates.config import waffle
+
+
+SWITCHES = waffle.waffle()
+
+
+def auto_certificate_generation_enabled():
+    return (
+        SWITCHES.is_enabled(waffle.SELF_PACED_ONLY) or
+        SWITCHES.is_enabled(waffle.INSTRUCTOR_PACED_ONLY)
+    )
+
+
+def auto_certificate_generation_enabled_for_course(course):
```

Comment:
Nit: As we'll see in the Clean-Code book soon, it's better to pass only the minimum of what's needed here.  So, passing through the pacing_type (or is_self_paced) instead of the entire course.

Reply:
Yeah, good point.